python/galois/lonestar/analytics/average_degree_connectivity.py

```diff
@@ -0,0 +1,142 @@
+            outDict[deg] = newvalue
+        else: 
+            #add key and value 
+            #create an array with no finite size
+            nid_array = []
+            #add nid to value array
+            nid_array.append(node_id)
+            #set nid array as this degrees value
+            out_dict[deg] = nid_array
+
+    return out_dict
+
+#method that creates a dictionary of all the in degrees seen in graph g
+#where key = degree and value = array of node_id's with that degree 
+def fill_graph_in_deg(graph: PropertyGraph):
+    #get in degree array
+    in_array = graph.get_node_property("in_degree_property")
+    in_dict = dict()
+
+    for node_id, deg in enumerate(in_array): 
+        if deg in in_dict.keys():
+            #update value array
+            #but first get value 
+            newvalue = in_dict.get(deg) 
+            #add nid to new value 
+            newvalue.append(node_id)
+            #set new value as this degrees value
+            in_dict[deg] = newvalue
+
+        else: 
+            #add key and value 
+            #create an array with no finite size 
+            nid_array = []
+            #add nid to value array
+            nid_array.append(node_id)
+            #set nid array as this degrees value
+            in_dict[deg] = nid_array 
+
+    return in_dict
+
+#method that calculates the average degree connectivity of a graph g
+#returns dictionary where key = degree and value = degree's average degree connectivity
+def get_avg_degconn(graph: PropertyGraph, source, target, nodes, weight):
+
+    calculate_degree(graph, in_degree_property="in_degree_property", out_degree_property="out_degree_property", weight_property= weight)
+
+    #ask about correct way to pass in parameters :)
+    in_dict = fill_graph_in_deg(graph)
+    out_dict = fill_graph_out_deg(graph)
+
+    #create map that will hold the results 
+    #where key = deg and value = avg_degconn
+    result_dict = dict()
+
+    if source == "in" and target == "in":
+
+        for degree in in_dict.keys(): 
+
+            #get value (array of NID's) that have that degree
+            value = in_dict.get(degree)
+
+            #the average degree connectivity of a degree in graph g is the sum 
+            #of the average neighbor degree for nodes with degree k divided by 
+            #the number of nodes with that degree
+            avg_degree_connectivity = avg_n_sum(value) / len(value)
+            
+            #add key value pair
+            result_dict[degree] = avg_degree_connectivity
+
+    elif source == "out" and target == "out":
+
+        for degree in out_dict.keys(): 
+
+            #get value (array of NID's)
+            value = outDict.get(degree)
+            avg_degree_connectivity = avg_n_sum(value) / len(value)
+
+            #add key value pair
+            result_dict[degree] = avg_degree_connectivity
+
+    for key, value in result_dict.items():
+        print(key, ':', value)
```

Comment:
Return the result instead of printing it. If you need to print the value write a driver program that calls the function.